ResearchKit/Common/ORKAnswerFormat.h

```diff
@@ -1185,6 +1188,33 @@ ORK_CLASS_AVAILABLE
 @interface ORKTextAnswerFormat : ORKAnswerFormat
 
 /**
+ Returns an initialized text answer format using the regular expression.
+ 
+ This method is one of the designated initializers.
+ 
+ @param expression                The regular expression used to valide the text.
+ @param description               The description of the valid input presented to the user upon invalid input.
+ 
+ @return An initialized validated text answer format.
+ */
+- (instancetype)initWithValidationExpression:(NSString *)expression
+                       validInputDescription:(NSString *)description NS_DESIGNATED_INITIALIZER;
+
+/**
+ The regex used to validate user's input.
+ 
+ The default value is nil. If set to nil, no validation will be performed.
+ */
+@property (readonly) NSString *regex;
+
+/**
+ The text presented to the user that describes what the valid input is.
+ 
+ The default value is nil.
+ */
+@property (readonly) NSString *invalidMessage;
+
+/**
```

Comment:
Can we move both the properties below the second initializer? It's nice to keep all the initializers stacked at the top and all the properties below them.